models/Performance.js

```diff
@@ -27,95 +27,37 @@ module.exports = class Performance {
   }
 
   static create(name, performDate, openAt, closeAt) {
-    return new Promise((resolve, reject) => {
-      const client = utils.db.createClient();
-      const queryString = 'INSERT INTO performances (name, performDate, openAt, closeAt) VALUES ($1, $2, $3, $4) RETURNING id, closeAt';
-      let perfId;
-
-      if (
-        Number.isNaN(Date.parse(openAt)) ||
-        Number.isNaN(Date.parse(closeAt)) ||
-        Number.isNaN(Date.parse(performDate))
-      ) {
-        reject();
-      } else {
-        client.connect();
-        client.on('error', err => reject(err));
-
-        const query = client.query(queryString, [name, performDate, openAt, closeAt]);
-
-        query.on('row', ({ id }) => {
-          perfId = id;
-        });
-
-        query.on('end', () => {
-          client.end();
-          resolve(new Performance(perfId, name, openAt, closeAt, performDate));
-        });
-
-        query.on('error', (err) => {
-          client.end();
-          reject(err);
-        });
-      }
-    });
+    const sql = 'INSERT INTO performances (name, performDate, openAt, closeAt) VALUES ($1, $2, $3, $4) RETURNING id, closeAt';
+
+    if (
+      Number.isNaN(Date.parse(openAt)) ||
+      Number.isNaN(Date.parse(closeAt)) ||
```

Comment:
This is an interesting pattern for date validation that I'd not seen before. I'll have to look into how it works later